### ISSUE TITLE
Creating visualizations failing - No such file or directory

### ISSUE THREAD

**Issue by squ1ddy** [atc-project/atomic-threat-coverage#154]
Hello,

Receiving this error every time the `docker run -it atc' command is run:

```
[*] Creating visualizations..
/app/scripts/atc_visualizations/yaml_handler.py:36: YAMLLoadWarning: calling yaml.load_all() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  return [x for x in yaml_fields]
/app/scripts/atc_visualizations/yaml_handler.py:511: YAMLLoadWarning: calling yaml.load_all() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  _ = [x for x in _]
Traceback (most recent call last):
  File "main.py", line 160, in <module>
    YamlHandler(dashboard_path + file, output_path +
  File "/app/scripts/atc_visualizations/yaml_handler.py", line 73, in __init__
    with open(output_file, 'w') as f:
FileNotFoundError: [Errno 2] No such file or directory: '../analytics/generated/visualizations/os_hunting_dashboard.json'
[*] Generating TheHive Case templates based on Response Playbooks
HINT: Make sure proper directories are configured in the config.yml
Done!
```

Indeed the file/folder does not exist:

```bash
scottm@caret:~/atomic-threat-coverage$ ls
analytics               detection_rules       images            README.md         response_playbooks  visualizations
Atomic_Threat_Coverage  DEVELOP.md            LICENSE           README_PL.md      run_tests.sh
CONTRIBUTING.md         docker-entrypoint.sh  logging_policies  README_RU.md      scripts
customers               Dockerfile            Makefile          requirements.txt  tests
data_needed             enrichments           mitigations       response_actions  triggers
scottm@caret:~/atomic-threat-coverage$ cd analytics/
scottm@caret:~/atomic-threat-coverage/analytics$ ls
generated  predefined
scottm@caret:~/atomic-threat-coverage/analytics$ cd generated/
scottm@caret:~/atomic-threat-coverage/analytics/generated$ ls
analytics.csv  atc_es_index.json  attack_navigator_profiles  pivoting.csv  thehive_templates
scottm@caret:~/atomic-threat-coverage/analytics/generated$
```

Any assistance resolving this would be much appreciated. Given this is the 3rd issue I've submitted I'm guessing that I'm doing something fundamentally wrong so apologies in advance and I'm going to stop submitting (there are several more errors during docker build). I have followed all the directions in readme.md.

Thanks in advance,

squ1ddy

**Comment by mrblacyk** [comment]
Hey,

I'm starting to look into it. I will let you know with my findings.

**Comment by mrblacyk** [comment]
I found the issue to be a very trivial one. Simply, the output directory did not exist out of the box. Now, I have added functionality of creating the directory/directories if they do not exist. Sorry this happened.

Tested it after the fix was implemented with freshly cloned repository and I confirm it solving the problem. 

Thanks for pointing this out and don't limit yourself with issues. It's more helpful for us and the community to have them and even close them with only an explanation instead of not having them at all.

Closing the issue. Take care!
